spec/integrations/food_index_spec.rb

```diff
@@ -0,0 +1,42 @@
+require 'rails_helper'
+
+RSpec.feature 'Food List Page', type: :feature do
+  before do
+    @user = User.create(name: 'Test User', email: 'test@gmail.com', password: 'password', confirmed_at: Time.now)
+
+    visit new_user_session_path
+
+    fill_in 'Email', with: @user.email
+    fill_in 'Password', with: @user.password
+    click_button 'Log in'
+
+    Food.create(name: 'Pizza', price: 10, measurement_unit: 'Slice', user_id: @user.id)
+    Food.create(name: 'Salad', price: 8, measurement_unit: 'Bowl', user_id: @user.id)
+
+    visit foods_path
+  end
+
+  scenario 'displays Add Food link for the current user' do
+    expect(page).to have_content('Add Food')
+  end
+
+  scenario 'redirects to the new food page when Add Food link is clicked' do
+    click_link 'Add Food'
+    expect(page).to have_content('Create a New Food')
+    expect(page).to have_current_path(new_food_path(@user))
+  end
```

Comment:
- [ ] You have done an amazing job so far. Unfortunately, some tests are failing, Please ensure all tests are passing and green.

![image](https://github.com/Xghayor/Recipe-craft/assets/68796232/3a63c539-e775-462b-8d0c-9fb9b4ee5f34)